Client/src/Components/NavigationMenu.tsx

```diff
@@ -1,46 +1,20 @@
 /* eslint-disable react/display-name */
-/* eslint-disable react/prop-types */
 
-import { AppBar, Box, Button, Divider, Drawer, Grid, IconButton, List, ListItem, ListItemIcon, Toolbar, Typography, makeStyles } from "@material-ui/core"
-import {
-  HashRouter,
-  Link,
-  NavLink,
-  Route
-} from "react-router-dom"
+import { AppBar, Box, Button, Divider, Drawer, Grid, IconButton, Toolbar, Typography, makeStyles } from "@material-ui/core"
+import { HashRouter, Link } from 'react-router-dom'
+import { ListRouter, getRoutedViews } from "./ListRouter"
 
-import { aboutRoute, dataCellAnalysisRoute, datasetUploadRoute, fileUploadRoute, graphRoute, homeRoute, profileRoute, researchPaperAnalysisRoute, searchRoute, signInRoute, signUpRoute} from '../Consts/Routes'
-
-
-import { AboutView } from "./Home/AboutView"
-import AccountBoxIcon from '@material-ui/icons/AccountBox'
-import BarChartIcon from '@material-ui/icons/BarChart'
 import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
-import CloudUploadIcon from '@material-ui/icons/CloudUpload'
-import { DataCellAnalysisView } from "./DataCellAnalysis/DataCellAnalysisView"
-import { DatasetUploadView } from "./DatasetUpload/DatasetUploadView"
-import DonutSmallIcon from '@material-ui/icons/DonutSmall'
-import FileUploadView from "./DataCell/FileUploadView"
-import GraphView from "./Graph/GraphView"
-import HomeIcon from '@material-ui/icons/Home'
-import HomeView from './Home/HomeView'
-import ImageSearchIcon from '@material-ui/icons/ImageSearch'
-import InfoIcon from '@material-ui/icons/Info'
 import MenuIcon from '@material-ui/icons/Menu'
-import { ProfileView } from "../Views/ProfileView"
 import React from 'react'
-import { ResearchPaperAnalysisView } from "./ResearchPaperAnalysis/ResearchPaperAnalysisView"
-import SearchIcon from '@material-ui/icons/Search'
-import SearchView from "./Search/SearchView"
 import clsx from "clsx"
+import { linkWidth } from './ListRouter'
+import { signInRoute } from "../Consts/Routes"
 import universitylogo from './universitylogo.png'
-import PersonIcon from '@material-ui/icons/Person';
-import SignIn from "../Authentication/SignIn"
-import SignUp from "../Authentication/SignUp"
 
-const drawerWidth = 240
+const drawerWidth = linkWidth
```

Comment:
Ok, so this is the main change. Also good job cleaning up the rest of the boilerplate code. This is what increases our good quality